LibGit2Sharp/Branch.cs

```diff
@@ -226,7 +236,17 @@ private Branch ResolveTrackedBranch()
                 return null;
             }
 
-            string trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            string trackedReferenceName = null;
+            try
+            {
+                trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            }
+            catch (InvalidSpecificationException)
+            {
+                // we have no way to pro-actively check this
```

Comment:
Okay, so I'm not sure this is going to work either.

We can sanitize the `CanonicalBranch` in the `BranchModel` ctor, but that means we can't read `libgit2branch.IsTracking` - because it'll evaluate this code path and we're back to square one...

I think the worst thing that is going to happen here is that tracking branches won't be acknowledged - and even if they have a valid `CanonicalName` we still need to catch the exception here for our friend with the wacky remote setup.

:cocktail: :man: :gun: